Fix analytics event documentation formatting

zachmargolis opened the pull request:
- "Email and Password Authentication" description was wrong
- Similar to #11640

**Before**

The description included some notes from the `new_device` attribute

```bash
> make analytics_events
> jq -r '.events[] | select( .event_name | contains("Email and Password") ) | .description' < public/api/_analytics-events.json
the attempt was unsuccessful, since it cannot be known whether it's a new device.
Tracks authentication attempts at the email/password screen
```

**After**

The description is only the description

```bash
> make analytics_events
> jq -r '.events[] | select( .event_name | contains("Email and Password") ) | .description' < public/api/_analytics-events.json
Tracks authentication attempts at the email/password screen
```

aduth reviewed (approved):
LGTM 👍 

Another idea for enforcing this: Maybe we can require that the method description starts uppercase? That might be enough of a hint to people in scenarios like this.

zachmargolis commented:
> Another idea for enforcing this: Maybe we can require that the method description starts uppercase? That might be enough of a hint to people in scenarios like this.

Oh that's a great idea! Let me give that a shot

zachmargolis commented:
> > Another idea for enforcing this: Maybe we can require that the method description starts uppercase? That might be enough of a hint to people in scenarios like this.
> 
> Oh that's a great idea! Let me give that a shot

Ok so after doing that, a few updates!
- Basic lowercase checking in ac7a2ec8f095d2996fdea1fe2e9a382bd69227e9
- Turns out `# rubocop:disable FooBar` lines were getting included in descriptions and flagging this error, so in 3a6c537b89010a92e3cbd0e2b3094e6c5e1e2e90 I updated to strip those out, which is a win for the resulting documentation too
- Fixed errors in 46fa55c76c48032f8f9222b775021e5422ec95fe, which included capitalizing some desciptions

The errors now look like

```
app/services/analytics_events.rb:7109 sms_opt_in_visit method description starts with lowercase, check indentation
```

which hopefully is clear to future us what we are trying to fix

aduth commented on lib/analytics_events_documenter.rb:
```diff
@@ -129,6 +129,11 @@ def missing_documentation
         errors << "#{error_prefix} #{tag.name} missing types" if !tag.types
       end
 
+      description = method_description(method_object)
+      if description.present? && !method_object.docstring.match?(/\A[A-Z]/)
+        errors << "#{error_prefix} method description starts with lowercase, check indentation"
+      end
```
Could we include the parsed description in the error message so it'll be even more obvious if the description includes param text? For context, I was previously unaware of the indentation requirement, so "check indentation" probably wouldn't have been enough for me to realize if a description includes parameter text.

zachmargolis replied:
How about 24e3453036c8554f5721723e5f41e1d286ff9ab7, which would look like this:

```
app/services/analytics_events.rb:249 add_email_confirmation method description starts with lowercase, check indentation:
  selection.
  A user has clicked the confirmation link in an email"
```

aduth replied:
wfm 👍

aduth reviewed (approved):
Big wins all around!
